### PR TITLE
PLAT-52544: Clean up samples

### PR DESCRIPTION
**Pull request by Baekwoo-s**
Clean up the items specified in DoD of the PLAT-52544.

Enact-DCO-1.0-Signed-off-by: Baekwoo Jung (baekwoo.jung@lge.com)

### CODE REVIEW

File: enact-all-samples/README.md [hunk 1]
```diff
@@ -1,3 +1,3 @@
 ### enact-all-samples
 
-This app is a wrapper app that links to all of the other sample apps.
\ No newline at end of file
+This app is a wrapper app that links to all of the other sample apps.
```
Comment (ybsung) [inline enact-all-samples/README.md]: Please check why this file was modified.

**Comment by ybsung** [comment]
In my opinion, it would be better to update the following names of the package.json files.

```
\samples\pattern-list-details-redux\package.json: "name": "CityViewer",
: The name would be `pattern-list-details-redux`.

\samples\pattern-list-details\package.json: "name": "CityViewer",
: The name would be `pattern-list-details`.

\samples\pattern-routable-panels\package.json: "name": "panel-routable-sample",
: The name would be `pattern-routable-panels`.

\samples\pattern-single-panel-redux\package.json: "name": "ProfilePhotoPicker",
: The name would be `pattern-single-panel-redux`.

\samples\pattern-single-panel\package.json: "name": "ProfilePhotoPicker",
: The name would be `pattern-single-panel`.

\samples\pattern-virtualgridlist-api\package.json: "name": "virtualgridlist-sample",
: The name would be `pattern-virtualgridlist-api`.
```

**Comment by ybsung** [comment]
How about updating `description` of the `package.json`?

\samples\pattern-list-details-redux\package.json:
~"description": "Sample list details Enact application.",~
"description": "A list details with redux sample Enact Moonstone application.",

\samples\pattern-list-details\package.json:
~"description": "Sample list details Enact application.",~
"description": "A list details sample Enact Moonstone application.",

\samples\pattern-virtualgridlist-api\package.json:
~"description": "A sample virtualgridlist with redux",~
"description": "A VirtualGridList with redux sample Enact Moonstone application.",

\samples\pattern-single-panel-redux\package.json:
~"description": "A sample single panel Enact application.",~
"description": "A sample single panel with redux Enact application.",

**Comment by ybsung** [comment]
In \samples\enact-all-samples\package.json,
It should be `"extends": "enact/strict"` instead of `"extends": "enact"`.

**Comment by ybsung** [comment]
It would be better to have the same title between package.json and appinfo.json. In addition, some samples do not have `webos-meta` folders.

path | package.json's title | appinfo.json's title
------------ | ------------- | ------------- | 
\samples\enact-all-samples\ | "Enact all samples" | "Enact QA Samples"
\samples\pattern-activity-panels-deep-linking\ | "Deep linking ActivityPanels sample" | "Activity Panels Deep Link Pattern"
\samples\pattern-activity-panels-redux\ | "ActivityPanels with redux sample" | "Activity Panels Pattern with Redux"
\samples\pattern-activity-panels\ | "ActivityPanels sample" | "Activity Panels Pattern"
\samples\pattern-dynamic-panel\ | "DynamicPanel sample" | "Enact App Template"
\samples\pattern-expandablelist-object\ | "ExpandableListObject sample" | "Enact App Template"
\samples\pattern-list-details-redux\ | "CityViewer sample redux" | **webos-meta folder missing**
\samples\pattern-list-details\ | "CityViewer sample" | **webos-meta folder missing**
\samples\pattern-locale-switching\ | "LocaleSwitching sample" | "Enact App Template"
\samples\pattern-ls2request\ | "ls2request sample" | "Enact LS2Request Sample"
\samples\pattern-routable-panels\ | "Routable sample" | **webos-meta folder missing**
\samples\pattern-single-panel-redux\ | "Profile Photo Picker" | **webos-meta folder missing**
\samples\pattern-single-panel\ | "Profile Photo Picker" | **webos-meta folder missing**
\samples\pattern-video-player\ | "VideoPlayer Pattern Sample" | "Activity Panels Pattern"
\samples\pattern-virtualgridlist-api\ | "VirtualGridList sample" | **webos-meta folder missing**
\samples\pattern-virtuallist-preserving-focus\ | "VirtualList preserving focus sample" | "Enact App Template"
\samples\tutorial-hello-enact\ | "HelloEnact sample" | "Hello, Enact"
\samples\tutorial-kitten-browser\ | "KittenBrowser sample" | "Kitten Browser"

**Comment by ybsung** [comment]
I don't know what is the best ids for appinfo.json files.

File: pattern-dynamic-panel/src/components/FileBrowser/FileBrowser.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+import Cancelable from '@enact/ui/Cancelable';
 import {Image} from '@enact/moonstone/Image';
 import {Item} from '@enact/moonstone/Item';
-import Cancelable from '@enact/ui/Cancelable';
-import {VirtualList} from '@enact/moonstone/VirtualList';
+import kind from '@enact/core/kind';
 import PropTypes from 'prop-types';
+import React from 'react';
+import ri from '@enact/ui/resolution';
+import {VirtualList} from '@enact/moonstone/VirtualList';
```
Comment (ybsung) [inline pattern-dynamic-panel/src/components/FileBrowser/FileBrowser.js]: `import VirtualList from '@enact/moonstone/VirtualList';` would be better.

**Comment by Baekwoo-s** [comment]
I have applied all the above comments. and I modified appinfo.json and package.json to be identical to other samples.

**Comment by ybsung** [comment]
Too many properties were omitted in `samples\package.json`. Could you check it?

**Comment by Baekwoo-s** [comment]
@ybsung 
React-router has been removed because it is an unused module. (using react-router-dom in samples)

**Comment by ybsung** [comment]
LGTM

**Comment by ybsung** [comment]
LGTM

File: enact-all-samples/package.json [hunk 3]
```diff
@@ -10,7 +10,8 @@
     "pack-p": "enact pack -p",
     "watch": "enact pack --watch",
     "clean": "enact clean",
-    "lint": "enact lint .",
+    "lint": "enact lint --strict .",
+    "license": "enact license",
```
Comment (webOS101) [inline enact-all-samples/package.json]: License should be Apache 2.0

File: enact-all-samples/src/components/SampleItem/SampleItem.js [hunk 4]
```diff
@@ -1,24 +1,26 @@
...
+		path: PropTypes.any
 	}
 
 	itemSelect = () => {
 		this.props.history.push({pathname: this.props.path});
 	}
 
 	render () {
-		const {children, rest} = this.props;
+		const {children, ...rest} = this.props;
```
Comment (webOS101) [inline enact-all-samples/src/components/SampleItem/SampleItem.js]: Oops!  Good catch.

File: pattern-activity-panels-deep-linking/package.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "pattern-activity-panels-deep-linking",
   "version": "1.0.0",
-  "description": "Deep linking ActivityPanels sample Enact Moonstone application.",
+  "description": "A sample activity panels with redux and deep linking Enact application.",
```
Comment (webOS101) [inline pattern-activity-panels-deep-linking/package.json]: Perhaps 'An Enact application demonstrating deep linking with ActivityPanels'?  I think it's useful to keep the component name being demonstrated in the description.

File: pattern-activity-panels-redux/package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "pattern-activity-panels-redux",
   "version": "1.0.0",
-  "description": "ActivityPanels with redux sample Enact Moonstone application.",
+  "description": "A sample activity panels with redux Enact application.",
```
Comment (webOS101) [inline pattern-activity-panels-redux/package.json]: As above:  'An Enact application demonstrating ActivityPanels with Redux'

File: tutorial-kitten-browser/webos-meta/appinfo.json [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
-	"type": "web",
-	"main": "index.html",
-	"title": "Kitten Browser",
-	"icon": "icon.png",
-	"miniicon": "icon-mini.png",
-	"largeIcon": "icon-large.png",
-	"uiRevision": 2
-}
\ No newline at end of file
+  "id": "tutorial-kitten-browser",
```
Comment (webOS101) [inline tutorial-kitten-browser/webos-meta/appinfo.json]: Would you please update all the `"id"`'s to be in the form:  `"com.enactjs...."`?

Comment (Baekwoo-s) [reply tutorial-kitten-browser/webos-meta/appinfo.json]: ok, I will update.

File: pattern-activity-panels-redux/package.json (hunk 6 above)
Comment (Baekwoo-s) [reply pattern-activity-panels-redux/package.json]: I will update description for all samples.

**Comment by Baekwoo-s** [comment]
I have applied all the above comments.

**Pull request merged**